src/main/java/com/aws/greengrass/shadowmanager/model/configuration/ShadowSyncConfiguration.java

```diff
@@ -89,14 +89,28 @@ public static ShadowSyncConfiguration processConfiguration(Map<String, Object> c
      * @param syncConfigurationSet the sync configuration list to add the device thing configuration to.
      * @throws InvalidRequestParametersException if the named shadow validation fails.
      */
-    private static void processOtherThingConfigurations(Map<String, Object> configTopicsPojo, String thingName,
+    private static void processOtherThingConfigurations(Map<String, Object> configTopicsPojo,
                                                         Set<ThingShadowSyncConfiguration> syncConfigurationSet) {
+        // Process the shadow documents map first.
+        configTopicsPojo.computeIfPresent(CONFIGURATION_SHADOW_DOCUMENTS_MAP_TOPIC,
+                (ignored, shadowDocumentsObject) -> {
+                    if (shadowDocumentsObject instanceof Map) {
+                        Map<String, Object> shadowDocumentsMap = (Map) shadowDocumentsObject;
+                        shadowDocumentsMap.forEach((componentName, componentConfigObject) ->
+                                processThingConfiguration(componentConfigObject, componentName, syncConfigurationSet));
+                    } else {
+                        throw new InvalidConfigurationException(String.format("Unexpected type in %s: %s",
+                                CONFIGURATION_SHADOW_DOCUMENTS_TOPIC, shadowDocumentsObject.getClass().getTypeName()));
+                    }
+                    return shadowDocumentsObject;
+                });
+
+        // Then process the shadow documents list.
```

Comment:
if we want the map to take priority should the map be processed last?

Reply:
That does not matter. Since we pass the set of `ThingShadowSyncConfiguration` to both the function. `ThingShadowSyncConfiguration` only contains the `thingName` and `shadowName`. So there is no chance of there being any conflicts with the map and the list.